### ISSUE TITLE
Embedded libtomcrypt 1.8.2 CVE-2019-17362 

### ISSUE THREAD

**Issue by robrwo** [atrodo/Net-Dropbear#6]
The dropbear library includes a version of libtomcrypt 1.8.2 that may have a security issue CVE-2019-17362

**Comment by atrodo** [comment]
Hello,

I have looked over this CVE. Net::Dropbear includes a full copy of Dropbear, as-is, in the distribution. I've looked at the current version of Dropbear, and it still includes v1.18.2 and I don't see any comments on the mailing list or repo about this vulnerability. Checking against the noted function, `der_decode_utf8_string`, Dropbear does not appear to use that function. Net::Dropbear uses Dropbear exclusively, it does not use anything from libtomcrypt directly, so this appears to indicate that Net::Dropbear is only as vulnerable as Dropbear is, and it appears that Dropbear isn't marked anywhere as vulnerable to it.

I will leave this issue for two purposes:

- [x] Check with the Dropbear to make sure that this vulnerability does not affect Dropbear
- [x] Update Net::Dropbear to include the newest version of Dropbear, currently 2022.82

**Comment by atrodo** [comment]
I have reviewed Dropbear's usage of libtomcrypt, and the function in question for CVE-2019-17362, `der_decode_utf8_string`, is not used in Dropbear. None of the DER parsing from libtomcrypt is used in Dropbear at all, I have confirmed that the flag to include it is not set, and confirmed that the resultant Dropbear.so that is built by Net::Dropbear does not include any of the `der_*` symbols. I have also just released Net::Dropbear v0.16, upgrading Dropbear to the newest version 2022.82.
